File: 2025-2/sesion-de-clase-25/src/Alumno.java
/**
 *
 * @author erichuiza
 */
public class Alumno extends Persona {
    private int codigo;
    private String facultad;
    private int ciclo;
    private boolean matriculado;
    
    public Alumno(int dni, String nombre, int edad, int codigo, 
            String facultad, int ciclo, boolean matriculado) {
        super(dni, nombre, edad);
        
        this.codigo = codigo;
        this.facultad = facultad;
        this.ciclo = ciclo;
        this.matriculado = matriculado;
    }
    
    public boolean isMatriculado() {
        return matriculado;
    }
    
    public void setMatriculado(boolean matriculado) {
        this.matriculado = matriculado;
    }
    
    public int getCodigo() {
        return codigo;
    }
    
    public String getFacultad() {
        return facultad;
    }
    
    public int getCiclo() {
        return ciclo;
    }
    
    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }
    
    public void setFacultad(String facultad) {
        this.facultad = facultad;
    }
    
    public void setCiclo(int ciclo) {
        this.ciclo = ciclo;
    }

    @Override
    public String toString() {
        return super.toString() + String.format(
                "Codigo: %d, Facultad: %s, "
                    + "Ciclo: %d, Matriculado: %b", 
                codigo, facultad, ciclo, matriculado);

//        return "Codigo: " + codigo + ", Nombre: " + getNombre() + 
//                ", Facultad: " + facultad + ", Ciclo: " + ciclo + 
//                ", Matriculado: " + matriculado;
    }    
}

File: 2026-1/capitulo-03/polimorfismo/ejercicio01/Alumno.cpp
//
// Created by Eric Huiza on 5/30/26.
//

#include "Alumno.h"

Alumno::Alumno() {
    codigo =  0;
    ciclo = 0;
}

Alumno::Alumno(int dni, int edad, int codigo, int ciclo) : Persona(dni, edad) {
    this->codigo = codigo;
    this->ciclo = ciclo;
}

int Alumno::getCodigo() const {
    return codigo;
}

int Alumno::getCiclo() const {
    return ciclo;
}

Alumno::~Alumno() {
    cout << "destructor del alumno." << endl;
}
